src/Sarif.Multitool/FileWorkItemsCommand.cs

```diff
@@ -132,25 +106,80 @@ private bool ValidateOptions(FileWorkItemsOptions options, IFileSystem fileSyste
             // We convert it to a URI with the ctor Uri(string, UriKind.RelativeOrAbsolute).
             // If it succeeds, we can assign the result to a Uri-valued property; if it fails,
             // we can produce a helpful error message.
-            options.ProjectUri = new Uri(options.ProjectUriString, UriKind.RelativeOrAbsolute);
 
-            if (!options.ProjectUri.IsAbsoluteUri)
+            valid &= ValidateProjectUri(options.ProjectUri, workItemFilingConfiguration);
+
+ 
+            valid &= options.ValidateOutputOptions();
+
+            if (!string.IsNullOrEmpty(options.OutputFilePath))
+            {
+                valid &= DriverUtilities.ReportWhetherOutputFileCanBeCreated(options.OutputFilePath, options.Force, fileSystem);
+            }
+
+            valid &= EnsureSecurityToken(options, workItemFilingConfiguration);
+
+            return valid;
+        }
+
+        private static bool ValidateProjectUri(string projectUriString, SarifWorkItemContext workItemFilingConfiguration)
+        {
+            if (string.IsNullOrEmpty(projectUriString) && workItemFilingConfiguration.ProjectUri == null)
+            {
+                // No project URI was provided via the --project-uri argument or as
+                // part of an input file specified via --configuration.
+                Console.Error.WriteLine(MultitoolResources.WorkItemFiling_NoProjectUriSpecified);
+                return false;
+            }
+
+            Uri projectUri = null;
+            if (!string.IsNullOrEmpty(projectUriString) && !Uri.TryCreate(projectUriString, UriKind.RelativeOrAbsolute, out projectUri))
             {
-                string optionDescription = DriverUtilities.GetOptionDescription<FileWorkItemsOptions>(nameof(options.ProjectUriString));
+                // A valid URI could not be created from the value '{0}' of the '{1}' option.
+                Console.Error.WriteLine(MultitoolResources.WorkItemFiling_ErrorUriIsNotLegal);
+                return false;
+            }
+
+            // Any command-line argument that's provided overrides values specified in the configuration.
```

Comment:
>configuration [](start = 91, length = 13)

So consider parsing the command line argument first and ignoring the (possibly faulty) value in the configuration file.

Reply:
I'm not sure of the value of doing work to allow the presence of a faulty argument in the configuration file. Change my mind.

---
In reply to: [381017141](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381017141) [](ancestors = 381017141)